### PR TITLE
DBZ-2614 Support for infinite timestamps

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2614

built on https://github.com/debezium/debezium/pull/1877

### CODE REVIEW

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java [hunk 1]
```diff
@@ -597,6 +603,34 @@ private long asEpochMicros(String timestamp) {
...
+        // written to database: 4713-12-31T23:59:59.999999 BC
+        // arrived from decoding plugin: -210831897600000001
+        // value from decoding plugin converted to Instant: -4712-12-31T23:59:59.999999Z - 1 year difference
+        // JDBC driver delivers Timestamp B.C.E. 4713-12-31T23:59:59.000+0100 that internally contains -210835184391000001 and translates to Instant -4712-11-23T22:59:59.999999Z
+        // The result is that JDBC driver and logical decoding plugin provides different values which moreover
+        // do not map to Java conversions
+        // It seems that JDBC driver always uses Gregorian calendar while Java Time API uses Julian
+        // Also it seems that further distortions could be introduced by timezone used so instead of matching
+        // against exact value the requested date should be smaller than -4712-11-01
+        final SchemaAndValueField.Condition largeNegativeTimestamp = (String fieldName, Object expectedValue, Object actualValue) -> {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java:617]: Lots of good explanation, but one key part is missing: what's the approach for dealing with this? I.e. does the assertion have some leniency built in?

**Review by gunnarmorling — APPROVED** [review]
@jpechane, LGTM, not sure why that fell through the cracks. One small comment inline, other than that, it only needs a rebase to current master. Thx!

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java:617]: @gunnarmorling Already rebased. The leniency is described at exactly tha last too lines of the comment

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java:617]: Thx!

**Comment by gunnarmorling** [comment]
LGTM, merging. Test failure is the known one tracked via https://issues.redhat.com/browse/DBZ-3107. Thanks, @jpechane!

**Pull request merged**